src/services/bookmarkService.js

```diff
@@ -0,0 +1,60 @@
+const { v4: uuidv4 } = require("uuid");
+const { db } = require("../db/db");
+const { bookmark } = require("../db/schema");
+const { eq } = require("drizzle-orm");
+const { bookmarkSchema } = require("../utils/joiSchema");
+const Joi = require("joi");
+
+const getAllBookmarks = async () => {
+  try {
+    const result = await db.select().from(bookmark);
+    return result;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const getBookmarkById = async (id) => {
+  try {
+    await Joi.string().uuid().required().validateAsync(id);
+    const result = await db.select().from(bookmark).where(eq(bookmark.id, id));
+    return result.length > 0 ? result[0] : null;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const createBookmark = async (title, url) => {
+  try {
+    await bookmarkSchema.validateAsync({ title, url });
```

Comment:
Here, it is good to have a logic that prevents duplication.

Reply:
thanks for the suggestion it is done